core/src/main/kotlin/com/strumenta/kolasu/traversing/Structurally.kt

```diff
@@ -15,7 +15,8 @@ fun Node.walk(): Sequence<Node> {
     return sequence {
         yield(this@walk)
         this@walk.walkChildren().forEach {
-            yieldAll(it.walk())
+            yield(it)
+            yieldAll(it.walkChildren())
```

Comment:
Doesn't this stop at the second level of children?

Reply:
You are right. I forgot to adapt the test. So, I am  mistaken.